source/include/VirtualInstrument.h

```diff
@@ -62,12 +62,23 @@ public :
 
 public:
     VirtualInstrument(ExecutionContextRef context, int clumps, TypeRef paramsType, TypeRef localsType);
+    ~VirtualInstrument()               { if (_viName) delete _viName; }
     TypeManagerRef TheTypeManager()     { return _typeManager; }
     TypedObjectRef Params()             { return _params; }
     TypedObjectRef Locals()             { return _locals; }
     TypedArray1D<VIClump>* Clumps()     { return _clumps; }
-    SubString VIName()                  { return _viName; }
-    void SetVIName(SubString s)         { _viName = s; }
+    SubString VIName()                  {
+        SubString s;
+        if (_viName)
+            s.AliasAssignLen(_viName, IntIndex(strlen(ConstCStr(_viName))));
+        return s;
+    }
+    void SetVIName(const SubString *s)   {
+        IntIndex len = s->Length();
+        _viName = (Utf8Char*)malloc(len+1);
```

Comment:
May be, you meant to do "new" since you are calling "delete" in the destructor ~VirtualInstrument().